### PR TITLE
Generate website with jstatic + DRY out templates + simplify Gruntfile

### PR DESCRIPTION
**Pull request by bewildergeist**
Purely a change in how the website is generated. There is no change in the generated markup (save for the addition of the current year to the footer, e.g. "© 2014 – 2015" rather than only "© 2014", easily reverted if need be).

Intended as "plumbing" prior to design updates and website improvements from #433 or elsewhere.

To that end, this PR contributes the following:

- Eradicates markup duplication between `index.html`, `api.html` and `guide.html`
- Removes the need for `grunt-contrib-clean` (and the `tempFolder`) and `grunt-replace`
- Replaces `grunt-md2html` with [`jstatic`](http://azeemarshad.in/posts/introducing_jstatic.html) which offers a much more flexible site generation workflow and more sophisticated templating ([`swig`](http://paularmstrong.github.io/swig/) out of the box, but that can be swapped out)
- Runs the various `.json`-config files through `swig` templating which makes it possible to generate more of their content from the base `package.json` (currently just the version number is inserted)
- Adds YAML front matter to Markdown files, which currently just duplicates the title, but gives us a flexible metadata mechanism for other properties (last updated, meta-description, tags/keywords etc.)

I hope you agree to the advantages of this. If breaking it up over several commits is better, let me know.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -3,78 +3,49 @@ module.exports = function (grunt) { // eslint-disable-line
...
+
 	var inputFolder = "./docs"
-	var tempFolder = "./temp"
-	var archiveFolder = "./archive"
+	var layoutFolder = inputFolder + "/layout"
 	var outputFolder = "../mithril"
+	var archiveFolder = "./archive"
+	var currentVersionArchiveFolder = archiveFolder + "/v" + pkg.version
 
 	var guide = [
```
Comment (impinball) [inline Gruntfile.js]: How much of this list is actually different beyond order?

File: Gruntfile.js [hunk 2]
```diff
@@ -3,78 +3,49 @@ module.exports = function (grunt) { // eslint-disable-line
...
+		"benchmarks",
 		"practices",
-		"refactoring",
-		"routing",
 		"tools",
-		"web-services"
+		"community"
 	]
 
 	var api = [
```
Comment (impinball) [inline Gruntfile.js]: Same question

File: Gruntfile.js (hunk 1 above)
Comment (bewildergeist) [inline Gruntfile.js]: I've removed `refactoring` which doesn't have a corresponding page. The rest is just re-ordering to match the navigation on the site (just for consistency, makes no practical difference). Should've split it over two commits to make that clearer.

File: Gruntfile.js (hunk 2 above)
Comment (bewildergeist) [inline Gruntfile.js]: I've removed `mithril.xhr` which doesn't have a corresponding page. The rest is just re-ordering to match the navigation on the site (just for consistency, makes no practical difference). Should've split it over two commits to make that clearer.

Comment (impinball) [inline Gruntfile.js]: Thanks

**Comment by isiahmeadows** [comment]
Closing due to inactivity. If you feel this should still be done, please feel free to file a new issue or pull request. :smile:

**Pull request closed without merging**